sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
+   *
+   */
+  struct Range
+  {
+    /**
+     * @brief The starting point of the HTTP Range.
+     *
+     */
+    int64_t Offset = 0;
+
+    /**
+     * @brief The size of the HTTP Range.
+     *
+     */
+    int64_t Length = 0;
+  };
```

Comment:
Should we add a ctor and validate offset/length are not negative? And that length isn't 0?

Reply:
What about equality/comparison operators?

Reply:
> Should we add a ctor and validate offset/length are not negative? And that length isn't 0?

Another case for pre-condition check, which we will discuss in tomorrow's guideline meeting.

Should we throw or crash if the input is invalid.

Reply:
> What about equality/comparison operators?

I cannot think of its being used anywhere. We can always add it later.

Reply:
> We can always add it later.

Sounds good to me.